### PR TITLE
Fix app is quitted right after executing installed app

### PR DESCRIPTION
**Pull request by simonhong**
local string's buffer is used after it is freed.

Close https://github.com/brave/brave-browser/issues/467

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.

## Test Plan:
Execute installed app from dmg

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions

### CODE REVIEW

**Comment by darkdh** [comment]
`strdup` is called at the end of `ProductDirNameForBundle` so why making early return here

File: patches/chrome-common-chrome_paths_mac.mm.patch [hunk 1]
```diff
@@ -21,7 +21,7 @@ index cca3bbb41ae25484c3e87c9aba09a8b07741e2b2..9e2676c79f09a565b5047af20bb82d44
 +    product_dir_name = "BraveSoftware/Brave-Browser";
```
[the remark this answers is not in the document]
Comment (bridiver) [reply patches/chrome-common-chrome_paths_mac.mm.patch]: this also isn't consistent with chrome_paths_linux which has an official_build flag (although I'm not sure why that isn't part of GetChannelSuffixForDataDir)

**Comment by bridiver** [comment]
@simonhong that's what we do in muon. I think we would probably add it to the existing plist fixup script

**Comment by simonhong** [comment]
@bridiver Yep, we can do that easily our own tweak script :)

**Comment by simonhong** [comment]
Yay! a lot of code and patch file are deleted.
Thanks for nice review!

**Pull request merged**